wiki/common-platforms/ros/ros-unsupported-os.md

```diff
@@ -0,0 +1,184 @@
+### 2. Installing ROS2 Humble from source on Ubuntu 20.04
+#### Set locale
+```
+locale  # check for UTF-8
+
+sudo apt update && sudo apt install locales
+sudo locale-gen en_US en_US.UTF-8
+sudo update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8
+export LANG=en_US.UTF-8
+
+locale  # verify settings
+```
+#### Add the ROS 2 apt repository
+```
+sudo apt install software-properties-common
+sudo add-apt-repository universe
+sudo apt update && sudo apt install curl -y
+sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+```
+#### Install development tools and ROS tools
+```
+sudo apt update && sudo apt install -y \
+  python3-flake8-docstrings \
+  python3-pip \
+  python3-pytest-cov \
+  ros-dev-tools
+```
+```
+python3 -m pip install -U \
+   flake8-blind-except \
+   flake8-builtins \
+   flake8-class-newline \
+   flake8-comprehensions \
+   flake8-deprecated \
+   flake8-import-order \
+   flake8-quotes \
+   "pytest>=5.3" \
+   pytest-repeat \
+   pytest-rerunfailures
+```
+
+#### Create a workspace and clone all repos
+```
+mkdir -p ~/ros2_humble/src
+cd ~/ros2_humble
+vcs import --input https://raw.githubusercontent.com/ros2/ros2/humble/ros2.repos src
+```
+
+#### Install dependencies using rosdep
+```
+sudo apt upgrade
+sudo rosdep init
+rosdep update
+rosdep install --from-paths src --ignore-src -y --skip-keys "fastcdr rti-connext-dds-6.0.1 urdfdom_headers"
+```
+
+#### Build the code in the workspace
+```
+cd ~/ros2_humble/
+colcon build --symlink-install
+```
+
+#### Set up your environment by sourcing the following file
+```
+. ~/ros2_humble/install/local_setup.bash
+```
+
+#### Try examples
+By running the following commands in separate terminals, you should see the talker saying that it’s Publishing messages and the listener saying I heard those messages. This verifies both the C++ and Python APIs are working properly.
+```
+. ~/ros2_humble/install/local_setup.bash
+ros2 run demo_nodes_cpp talker
+```
+```
+. ~/ros2_humble/install/local_setup.bash
+ros2 run demo_nodes_py listener
+```
+## Summary
+In this article, we discussed how to run a ROS distribution on an unsupported OS. Specifically we looked at the process of running ROS2 Humble on Ubuntu 20.04. However, the methods presented in the article can be easily adapted to accommodate other scenarios.
+
+## See Also:
+- https://roboticsknowledgebase.com/wiki/tools/docker/
+
+## Further Reading
+- https://docs.docker.com/engine/reference/run/
+
+## References
+- https://docs.ros.org/en/humble/Installation/Alternatives.html
+- https://github.com/osrf/docker_images
+- https://hub.docker.com/_/ros/
```

Comment:
'See Also' and 'Further Reading' are redundant sections and can be merged.

For each link below, please add a descriptive title and link using hyperlinks instead of raw URLs.

Reply:
Added this change